src/named_source.rs

```diff
@@ -27,11 +27,6 @@ impl NamedSource {
         }
     }
 
-    /// Gets the name of this `NamedSource`.
-    pub fn name(&self) -> &str {
```

Comment:
I didn't notice this, but this is a breaking change because `SourceCode::name` has to be an Option<&str> but `NamedSource::name` was just `&str`. Should we keep this for compatibility even though it shadows the new `SourceCode` method?